Automattic/newspack-blocks
# feat: author list block

### All Submissions:






### Changes proposed in this Pull Request:



Registers a new Author List block that uses the same template as the existing Author Profile block to display all WP users, guest authors, or both. Also adds a few new display features that apply to both blocks.

Closes #901.

### How to test the changes in this Pull Request:

1. Ensure your site has a number of WP users and CAP guest authors, some assigned to posts, others with 0 posts. Ensure that you also have some WP users of various roles, e.g. editor, admin, author, contributor, and subscriber, as well as some custom roles (such as the WooCommerce user roles).
2. Test the new options for the Author Profile block. Add one to a post or page and test the following new options in both editor and front-end:
  - **Text size:** controls the relative size of the text compared to the surrounding post. Should offer S, M, L, and XL options with M being the default.
  - **Hide default avatars:** If the "display avatars" option is enabled, enabling this option will only display avatars for authors that have a custom avatar image.

3. Test the new Author List block. Test the various attributes and display options available, paying special attention to:
  - **Author Type:** CAP guest authors, WP users, or both.
  - **WP user roles:** If the Author Type contains WP users, this should default to core WP user roles that have the `edit_posts` capability (Admin, Editor, Author, and Contributor) but it should have additional options to display other custom user roles registered with that capability.
  - **Column view:** In the block toolbar (not the sidebar). Default view is a simple list, but enabling columns will display the list in columns and enable the Columns attribute.
  - **Columns:** In the sidebar. If column view is enabled, should let you specify 2–6 columns.
  - **Display alphabetical separators:** If enabled, letter headings will separate the authors shown based on their display name (first letter of last name, or first letter of name if the display name is only one word).
  - **Group authors by alphabet:** If column view and "display alphabetical separators" are enabled, this option should be available. If enabled, it will break the authors into sections by alphabetical separator, so that each section becomes its own set of columns.
  - **Author List Exclusions:** Options to exclude authors who have 0 posts as well as specific authors by name.
  - **Author Profile Settings:** Should have the same options and behavior as these settings in the Author Profile block. Settings will apply to all items in the block.
  - **Avatar Settings:** Should have the same options and behavior as these settings in the Author Profile block. Settings will apply to all items in the block.

4. Also test after disabling the Co-Authors Plus plugin—all guest-author related options should be disabled/hidden and the block should only show WP users.

### Other information:
